transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,213 @@
+	}
+	return "", fmt.Errorf("could not find the main function in python files")
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+	pythonFiles, err := common.GetFilesByExt(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilePath, err := findMainScript(pythonFiles)
+	if err != nil {
+		return nil, err
+	}
+	// Search requirements.txt
+	if mainPythonFilePath != "" {
+		serviceName := strings.TrimSuffix(filepath.Base(mainPythonFilePath), pythonExt)
+		if serviceName == pythonMain {
+			serviceName = pythonService
+		} else if serviceName == pythonManage {
+			serviceName = djangoService
+		}
+		services = map[string][]transformertypes.Artifact{
+			serviceName: {{
+				Paths: map[string][]string{
+					artifacts.ProjectPathPathType: {dir},
+					MainPythonFilePathType:        {mainPythonFilePath},
+				},
+			}},
+		}
+		return services, nil
+	}
+	return nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PythonDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var pythonConfig PythonTemplateConfig
+		pythonConfig.AppName = a.Name
+		if a.Name == djangoService {
+			pythonConfig.IsDjangoProject = true
+		}
+		mainPythonFileRelPath, err := filepath.Rel(a.Paths[artifacts.ProjectPathPathType][0], a.Paths[MainPythonFilePathType][0])
+		if err != nil {
+			logrus.Errorf("Error in getting the relative path %s", err)
+		} else {
+			pythonConfig.MainScriptRelPath = mainPythonFileRelPath
+		}
+		var detectedPorts []int32
+		detectedPorts = append(detectedPorts, 8080) //TODO: Write parser to parse and identify port
+		pythonConfig.Port = commonqa.GetPortForService(detectedPorts, a.Name)
+		requirementsTxtFiles, err := common.GetFilesByName(a.Paths[artifacts.ProjectPathPathType][0], []string{requirementsTxtFile}, nil)
```

Comment:
We can detect all these as part of the detect function as pass it as part of the artifact.

Reply:
Done.